src/backend/discovery/discovery.lambda.ts

```diff
@@ -0,0 +1,291 @@
+      console.log(`Received a batch of ${batch.length} length`);
+      const batchPromises = batch
+        // ignores changes which are not package update, see https://github.com/cdklabs/construct-hub/issues/3#issuecomment-858246275
+        .filter(change => change.doc.name)
+        .map(change => {
+          return processPackageUpdate(change)
+            .catch(err => {
+              Object.defineProperty(change, '_construct_hub_failure_reason', err);
+              return s3.putObject({
+                Bucket: stagingBucket,
+                Key: `${FAILED_CHANGE_PREFIX}/${change.id}`,
+                Body: JSON.stringify(change, null, 2),
+                Metadata: {
+                  'Lambda-Log-Group': functionContext.logGroupName,
+                  'Lambda-Log-Stream': functionContext.logStreamName,
+                  'Lambda-Run-Id': functionContext.awsRequestId,
+                },
+              }).promise().then(() => (void undefined));
+            });
+        });
+
+      await Promise.all(batchPromises).then(function() {
+        // write the last sequence to the marker file in S3
+        const lastSequence = batch[batch.length - 1].seq;
+        return writeMarkerToS3(lastSequence);
+      }).then(function() {
+        // resume reader
+        (db.changesReader as any).resume();
+      }).catch(function (error) {
+        throw Error(`Error while processing batch, marker will not be updated, exiting.\n ${error}`);
+      });
+    })
+    .on('end', () => {
+      console.log('Changes feed monitoring has stopped');
+    });
+}
+
+async function writeMarkerToS3(sequence: Number) {
+  const client = new AWS.S3();
+  await client.putObject({
+    Bucket: stagingBucket,
+    Key: MARKER_FILE_NAME,
+    Body: sequence.toString(),
+    Metadata: {
+      'Lambda-Log-Group': functionContext.logGroupName,
+      'Lambda-Log-Stream': functionContext.logStreamName,
+      'Lambda-Run-Id': functionContext.awsRequestId,
+    },
+  }).promise();
+}
+
+async function processPackageUpdate(change: Change) {
+  let latestVersion: VersionInfo;
+  try {
+    // we assume that the change is concerning the latest version
+    latestVersion = getLatestVersion(change);
+    if (!isJsiiModule(latestVersion) || !isConstruct(latestVersion)) {
+      return;
+    }
+  } catch (error) {
+    console.error(`Could not parse change for ${change.id}`);
+    throw error;
+  }
+
+  return new Promise<void>((ok, ko) => {
+    // change.dist.tarball => https://registry.npmjs.org/<@scope>/<name>/-/<name>-<version>.tgz
+    // staging bucket key  => packages/<@scope>/<name>/-/<name>-<version>.tgz
+    const key = latestVersion.dist.tarball.replace('https://registry.npmjs.org', 'packages');
+    copyPackageToS3(latestVersion, key).then(() => {
+      return sendSqsMessage(latestVersion, key);
+    }).catch(function(reason) {
+      console.error(`Error processing change for ${change.id}:\n ${reason}`);
+      ko();
+    });
```

Comment:
You don't need an explicit promise here since you're in an `async` function. 

```suggestion
  // change.dist.tarball => https://registry.npmjs.org/<@scope>/<name>/-/<name>-<version>.tgz
  // staging bucket key  => packages/<@scope>/<name>/-/<name>-<version>.tgz
  const key = latestVersion.dist.tarball.replace('https://registry.npmjs.org', 'packages');
  await copyPackageToS3(latestVersion, key)
  await sendSqsMessage(latestVersion, key);
```